nmap.cc

```diff
@@ -1727,6 +1675,77 @@ void  apply_delayed_options() {
   }
   o.exclude_spec = delayed_options.exclude_spec;
 
+  if (delayed_options.decoy_arguments) {
+    char *p = delayed_options.decoy_arguments, *q;
+    do {
+      q = strchr(p, ',');
+      if (q)
+        *q = '\0';
+      if (!strcasecmp(p, "me")) {
+        if (o.decoyturn != -1)
+          fatal("Can only use 'ME' as a decoy once.\n");
+        o.decoyturn = o.numdecoys++;
+      } else if (!strcasecmp(p, "rnd") || !strncasecmp(p, "rnd:", 4)) {
+        int i = 1;
+
+        /* 'rnd:' is allowed and just gives them one */
+        if (strlen(p) > 4)
+          i = atoi(&p[4]);
+
+        if (i < 1)
+          fatal("Bad 'rnd' decoy \"%s\"", p);
+
+        if (o.numdecoys + i >= MAX_DECOYS - 1)
+          fatal("You are only allowed %d decoys (if you need more redefine MAX_DECOYS in nmap.h)", MAX_DECOYS);
+
+        while (i--) {
+          do {
+            if (delayed_options.af == AF_INET6){
+              u8 addr[16];
+              get_random_ipv6(addr);
```

Comment:
I feel we should be more conservative for now: fatal error if `rnd` is used with IPv6, and a TODO comment to update it when/if we do `-iR` for IPv6. The reason: IPv6 addresses have "scope," so we can't just randomly assign one and assume it will work. It's a little more complicated than just checking reserved and private IPs like we do for IPv4. Instead of trying to solve that whole problem, let's get a mostly-functional `-D` for IPv6, which is what this PR is for.